ext/pgsql/pgsql.c

```diff
@@ -992,17 +1024,23 @@ PHP_FUNCTION(pg_query)
 
 	pgsql = link->conn;
 
-	if (PQsetnonblocking(pgsql, 0)) {
-		php_error_docref(NULL, E_NOTICE,"Cannot set connection to blocking mode");
-		RETURN_FALSE;
-	}
-	while ((pgsql_result = PQgetResult(pgsql))) {
-		PQclear(pgsql_result);
-		leftover = 1;
-	}
-	if (leftover) {
-		php_error_docref(NULL, E_NOTICE, "Found results on this connection. Use pg_get_result() to get these results first");
+#ifdef LIBPQ_HAS_PIPELINING
```

Comment:
A macro or a inline for this whole "repeated all over" block might be nice.

Reply:
I just added to the existing code.

It also seemed to me that there were a lot of repeating blocks.

But, I don't plan to optimize other people's code.

Reply:
it is not so much about optimising but more about factoring, if we update again we won t need to do it more than once.